toolkit/tools/imager/imager.go

```diff
@@ -328,50 +335,81 @@ func setupRootFS(outputDir, installRoot string) (extraMountPoints []*safechroot.
 	return
 }
 
-func setupDisk(outputDir, diskName string, liveInstallFlag bool, diskConfig configuration.Disk, rootEncryption configuration.RootEncryption, readOnlyRootConfig configuration.ReadOnlyVerityRoot) (diskDevPath string, partIDToDevPathMap, partIDToFsTypeMap map[string]string, isLoopDevice bool, encryptedRoot diskutils.EncryptedRootDevice, readOnlyRoot diskutils.VerityDevice, err error) {
+func setupDisks(outputDir string, diskNames []string, liveInstallFlag bool, disks []configuration.Disk, rootEncryption configuration.RootEncryption, readOnlyRootConfig configuration.ReadOnlyVerityRoot) (diskDevPaths []string, partIDToDevPathMap, partIDToFsTypeMap map[string]string, isLoopDevice bool, encryptedRoot diskutils.EncryptedRootDevice, readOnlyRoot diskutils.VerityDevice, err error) {
 	const (
 		realDiskType = "path"
 	)
-	if diskConfig.TargetDisk.Type == realDiskType {
+	var (
+		DiskType = realDiskType
+	)
+
+	logger.Log.Infof("configs (%d)", len(disks))
+
+	for i, disk := range disks {
+		if i == 0 {
+			DiskType = disk.TargetDisk.Type
+			continue
+		}
+		if disk.TargetDisk.Type != DiskType {
+			err = fmt.Errorf("Only one disk type can be specified found (%s) and (%s)", DiskType, disk.TargetDisk.Type)
+			return
+		}
+
+	}
+
+	if DiskType == realDiskType {
 		if liveInstallFlag {
-			diskDevPath = diskConfig.TargetDisk.Value
-			partIDToDevPathMap, partIDToFsTypeMap, encryptedRoot, readOnlyRoot, err = setupRealDisk(diskDevPath, diskConfig, rootEncryption, readOnlyRootConfig)
+			partIDToDevPathMap, partIDToFsTypeMap, encryptedRoot, readOnlyRoot, err = setupRealDisks(diskNames, disks, rootEncryption, readOnlyRootConfig)
```

Comment:
Does `diskNames` contain the actual device path here (i.e. `/dev/sda`)? Might be missing the translation step. If not, a comment/rename of variable might be in order.

Reply:
diskNames contains a list of names for disks that we need to create "disk#.raw", these are initally placed in defaultTempDiskNames. Would something like rawFileNames be better here?

Reply:
I believe this path is used only for ISO installers etc., where it's meant to target an actual physical disk (ie diskDevPath, eg /dev/sda).

Reply:
Storing them in a lookup could work, although isn't the path part of the disk object...? Might be better to refactor and just pull from there unless I'm missing something (`diskDevPath = diskConfig.TargetDisk.Value`)

Reply:
Ah i see what you are saying. i deleted the translation step there and did not add back an equivalent. in the next commit I will have a refactor that gets the correct value for diskdevpaths for real disks not using loopback